### PR TITLE
Changes to support the new RC4 format of JSONAPI.

### PR DESCRIPTION
**Pull request by lgebhardt**
Also updates the readme with an example for returning the total record count in the top level meta data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -709,17 +768,49 @@ JSONAPI::ResourceSerializer.new(PostResource).serialize_to_hash(PostResource.new
 
 This returns results like this:
 
-```ruby
+```json
```
Comment (brettchalupa) [inline README.md:771]: I believe for this code block to highlight nicely the content would need to be valid JSON, for example:

``` json
{
  "data": {
    "type": "posts",
    "id": 1
  }
}
```

You could drop the raw markdown into IRB and convert it to JSON to make it quick. :smile:

Comment (brettchalupa) [inline README.md:771]: @lgebhardt I converted it for you:

``` json
{
  "data": {
    "type": "posts",
    "id": "1",
    "links": {
      "self": "http://example.com/posts/1"
    },
    "attributes": {
      "title": "New post",
      "body": "A body!!!",
      "subject": "New post"
    },
    "relationships": {
      "section": {
        "links": {
          "self": "http://example.com/posts/1/links/section",
          "related": "http://example.com/posts/1/section"
        },
        "data": null
      },
      "author": {
        "links": {
          "self": "http://example.com/posts/1/links/author",
          "related": "http://example.com/posts/1/author"
        },
        "data": {
          "type": "people",
          "id": "1"
        }
      },
      "tags": {
        "links": {
          "self": "http://example.com/posts/1/links/tags",
          "related": "http://example.com/posts/1/tags"
        }
      },
      "comments": {
        "links": {
          "self": "http://example.com/posts/1/links/comments",
          "related": "http://example.com/posts/1/comments"
        }
      }
    }
  }
}
```

Feel free to copy the comment! :+1:

Comment (lgebhardt) [inline README.md:771]: Thanks! I need to look into a better markdown preview tool.

**Comment by john-griffin** [comment]
Tests pass with this branch on our API after updating them to RC4 syntax.

**Comment by dgeb** [comment]
Looks good @lgebhardt! Great work getting JR to 1.0 compliance so soon :+1:

**Comment by pixelhandler** [comment]
@lgebhardt thanks! updating my app today :)

**Pull request merged**